### PR TITLE
Use dialer for pipeline

### PR DESCRIPTION
**Pull request by gitlawr**
1. use dialer to connect Jenkins in user cluster, for https://github.com/rancher/rancher/issues/12245
2. filter resources by cluster in syncers to fix multiple instances issue.

### CODE REVIEW

File: pkg/controllers/user/pipeline/engine/engine.go [hunk 1]
```diff
@@ -17,19 +19,29 @@ type PipelineEngine interface {
...
-	nodeLister := cluster.Core.Nodes("").Controller().Lister()
 	serviceLister := cluster.Core.Services("").Controller().Lister()
 	secrets := cluster.Core.Secrets("")
 	secretLister := secrets.Controller().Lister()
 	managementSecretLister := cluster.Management.Core.Secrets("").Controller().Lister()
 	sourceCodeCredentialLister := cluster.Management.Management.SourceCodeCredentials("").Controller().Lister()
+
+	dial, err := cluster.Management.Dialer.ClusterDialer(cluster.ClusterName)
+	if err != nil {
+		logrus.Warnf("Failed to get cluster dialer: %v", err)
```
Comment (alena1108) [inline pkg/controllers/user/pipeline/engine/engine.go]: hm, and we just proceed if the error happens, and set the dial to nil on line 34? Is it expected behavior?

Comment (gitlawr) [reply pkg/controllers/user/pipeline/engine/engine.go]: Looks like clusterDialer can be a nodeDialer and may be invalid over time.  I think we should get the dial before doing the HTTP connection, not on controller registration. And we can check and handle the error there

**Comment by gitlawr** [comment]
@ibuildthecloud You mentioned that creating `http.Transports` on demand causes connection leaks, but I found that `cluterDialer()` returns a `nodeDialer`. I assume it will be invalid if the node is removed from the cluster?
>https://github.com/rancher/rancher/pull/12022#discussion_r176497834

cc @zionwu

File: pkg/controllers/user/pipeline/engine/engine.go (hunk 1 above)
Comment (gitlawr) [reply pkg/controllers/user/pipeline/engine/engine.go]: updated

**Comment by ibuildthecloud** [comment]
@gitlawr cluster dialer will return a node dialer if the cluster dialer can not be found.  But this is done dynamically on each Dial call, not once.  So a transient failure is possible during provision events but it will eventually work again.

**Comment by gitlawr** [comment]
I see, thanks

**Comment by gitlawr** [comment]
@alena1108  In that case, the pr should be good to go then.

**Pull request merged**